jormungandr/src/explorer/mod.rs

```diff
@@ -216,13 +216,38 @@ impl ExplorerDB {
                 )))
             }
         };
-        stream
+
+        let mut db = stream
             .map_err(Error::from)
             .try_fold(bootstraped_db, |mut db, block| async move {
                 db.apply_block(block).await?;
                 Ok(db)
             })
-            .await
+            .await?;
+
+        for branch in blockchain.branches().branches().await.iter() {
+            let mut hash = branch.hash();
+            let mut blocks = vec![];
+            loop {
+                if db.get_block(&hash).await.is_some() {
```

Comment:
I would just use `apply_block` and see if it returns the `AncestorNotFound` error. Otherwise I would use directly the multiverse lookup, as `get_block` only looks in the current explorer branch (although that will change with #2801), which I think may lead to applying a block twice.

Reply:
> as `get_block` only looks in the current explorer branch

Thanks for noting that. Yeah, for some branching cases this may result into applying a block twice. But will it break anything apart from the performance? Anyway, these branches are rather short (less than 10 blocks) and I wouldn't care much for the performance here. This is a procedure that is ran onle once and we are going to drop it soon.

Reply:
Hmm, for some reason I actually thought `apply_block` was going to throw an error, but now I think you are right, it will only affect performance. So yeah, it's fine to leave it as it is I guess.